crates/bevy_asset/src/io/metadata.rs

```diff
@@ -0,0 +1,58 @@
+/// A enum representing a type of file.
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum FileType {
+    Directory,
+    File,
+    // To be compatible with [`std::fs::FileType`].
+    Symlink,
+}
+
+impl FileType {
+    pub fn is_directory(&self) -> bool {
+        *self == Self::Directory
+    }
+
+    pub fn is_file(&self) -> bool {
+        *self == Self::File
+    }
+}
+
+impl From<std::fs::FileType> for FileType {
+    fn from(file_type: std::fs::FileType) -> Self {
+        if file_type.is_dir() {
+            Self::Directory
+        } else if file_type.is_file() {
+            Self::File
+        } else if file_type.is_symlink() {
+            Self::Symlink
+        } else {
+            unreachable!()
+        }
+    }
+}
+
+/// Metadata information about a file.
+///
+/// This structure is returned from the [`AssetIo::get_metadata`] method.
+#[derive(Debug)]
```

Comment:
```suggestion
#[derive(Debug, Clone)]
```
This should also derive `Clone`